src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -194,8 +194,12 @@ def vectorized_choice(
     index = pd.Index(np.arange(n_to_choose))
     if randomness_stream is None:
         # Generate an additional_key on-the-fly and use that in randomness.random
-        additional_key = f"{additional_key}_{random_seed}"
-        probs = random(str(additional_key), index)
+        # additional_key = f"{additional_key}_{random_seed}"
+        # probs = random(str(additional_key), index)
+        random_state = np.random.RandomState(seed=get_hash(f"{additional_key}_{random_seed}"))
+        sample_size = index.max() + 1
+        raw_draws = random_state.random_sample(sample_size)
+        probs = pd.Series(raw_draws[index], index=index)
```

Comment:
If the index is something like `pd.Index([1_000_000, 1_000_001])` we will do 1,000,002 samples to get two numbers we'll use. This seems quite wasteful. Why not do this?
```
raw_draws = random_state.random_sample(len(index))
probs = pd.Series(raw_draws, index=index)
```

Reply:
That makes sense. I did just copy this from vivarium - though I think in that case it works b/c you're dealing with a state table with a range index?